src/models/neural_networks/ealstm.py

```diff
@@ -0,0 +1,281 @@
+        if self.features_per_month is None:
+            assert x_ref is not None, \
+                f"x_ref can't be None if features_per_month or current_size is not defined"
+            self.features_per_month = x_ref[0].shape[-1]
+        if self.experiment == 'nowcast':
+            if self.current_size is None:
+                assert x_ref is not None, \
+                    f"x_ref can't be None if features_per_month or current_size is not defined"
+                self.current_size = x_ref[2].shape[-1]
+
+        return EALSTM(features_per_month=self.features_per_month,
+                      dense_features=self.dense_features,
+                      hidden_size=self.hidden_size,
+                      rnn_dropout=self.rnn_dropout,
+                      dense_dropout=self.dense_dropout,
+                      include_pred_month=self.include_pred_month,
+                      experiment=self.experiment,
+                      current_size=self.current_size)
+
+
+class EALSTM(nn.Module):
+    def __init__(self, features_per_month, dense_features, hidden_size,
+                 rnn_dropout, dense_dropout, include_pred_month,
+                 experiment, current_size=None):
+        super().__init__()
+
+        self.experiment = experiment
+        self.include_pred_month = include_pred_month
+
+        self.dropout = nn.Dropout(rnn_dropout)
+        self.rnn = EALSTMCell(input_size_dyn=features_per_month,
+                              input_size_stat=2,  # 2 for latlons, currently
+                              hidden_size=hidden_size,
+                              batch_first=True)
+        self.hidden_size = hidden_size
+        self.rnn_dropout = nn.Dropout(rnn_dropout)
+
+        dense_input_size = hidden_size
+        if include_pred_month:
+            dense_input_size += 12
+        if experiment == 'nowcast':
+            assert current_size is not None
+            dense_input_size += current_size
+        dense_features.insert(0, dense_input_size)
+
+        self.dense_layers = nn.ModuleList([
+            LinearBlock(in_features=dense_features[i - 1],
+                        out_features=dense_features[i],
+                        dropout=dense_dropout)
+            for i in range(1, len(dense_features))
+        ])
+
+        self.final_dense = nn.Linear(in_features=dense_features[-1], out_features=1)
+
+        self.initialize_weights()
+
+    def initialize_weights(self):
+
+        for dense_layer in self.dense_layers:
+            nn.init.kaiming_uniform_(dense_layer.linear.weight.data)
+
+        nn.init.kaiming_uniform_(self.final_dense.weight.data)
+        nn.init.constant_(self.final_dense.bias.data, 0)
+
+    def forward(self, x, pred_month=None, latlons=None, current=None):
+
+        # has to be optional to respect the positionality of the neural
+        # nets
+        assert latlons is not None, "Latlons can't be None!"
```

Comment:
can we not make latlons a required argument? Or  does the position change make that not possible?

Reply:
yea - it would screw up the indexing